lms/djangoapps/instructor/views/instructor_dashboard.py

```diff
@@ -65,8 +66,33 @@ def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument,redef
         return user and has_access(user, 'staff', course, course.id)
 
 
+def check_staff_or_404():
```

Comment:
Is this a new security check, independent of `sudo_required`? I'm surprised that this is not already implemented across both instructor dashboards.